### DISCUSSION TITLE
Batch Conversion from PDF to TIFF Pyramid

### DISCUSSION THREAD

**Discussion by JoshuaAPhillips** [ImageMagick/ImageMagick#6224] [Help] [answered]
### ImageMagick version

7.1.1

### Operating system, version and so on

MacOS m1 Ventura

### Description

Hello all,

Excuse a question that might be somewhat basic, but is there a way to batch produce pyramidal .tiff files using ImageMagick. I've several hundred PDF files that I need to convert to pyramidal .tiffs, and while I've had luck converting .pdf files to regular .tiff files using the following CLI command:

`for i in *; do convert -density 300 $i ./output/$i.tif; done`

... I can't seem to work out how to generate pyramidal .tiffs at scale, whether through modifying this command to something along the lines of 

`for i in *; do convert -density 300 $i -define tiff:tile-geometry=256x256 ./output/'ptif:$i.tif'; done`

... or a second command. I might be missing something very obvious (perhaps to do with the ptif: tag?) but would really appreciate your help.

Thanks and all best,
Joshua

**Comment by snibgo** [comment, marked as answer]
There is a problem in your output filename:
```
./output/'ptif:$i.tif'
```
The prefix `ptif:` should occur before the full path, not after the directory.

This would be wrong: `./output/'ptif:myfile.tif'`.

This would be correct: `'ptif:./output/myfile.tif'`.

**Comment by JoshuaAPhillips** [comment]
Hi @snibgo,

That seems to have worked: thank you so much for your help! 

For reference, this is the working command:

```
for i in *; do 
    convert -density 300 $i -define tiff:tile-geometry=256x256 ptif:./output/$i.tif; 
done
```

All best,
Joshua
